website/src/controllers/groups.js

```diff
@@ -137,7 +137,25 @@ api.get = function(req, res, next) {
   q.exec(function(err, group){
     if (err) return next(err);
     if (!group && gid!=='party') return res.json(404,{err: "Group not found or you don't have access."});
-    res.json(group);
+
+    //Since we have a limit on how many members are populate to the group, we want to make sure the user is always in the group
+    var userInGroup = _.find(group.members, function(member){ return member._id == user._id; });
+
+    //If the group is private or the group is a party, then the user must be a member of the group based on access restrictions above
+    if (group.privacy === 'private' || gid === 'party') {
+      //If the user is not in the group query, add them
+      if (userInGroup === undefined) { group.members.push(user); }
```

Comment:
Lodash uses semver so we can expect no breaking changes between 2.4 and let's say 2.19 so until we upgrade it we are going to be ok and anyway i think we would read the changelog when upgrading to a major release.

In this case, since the array won't contain any falsy value we can use `!` without problems as you said. But really for me both solutions are ok and I don't have any strong argument toward any of it